examples/hello_compute_c/main.c

```diff
@@ -0,0 +1,135 @@
+#include "./../../ffi/wgpu.h"
+#include <stdio.h>
+#include <stdlib.h>
+
+#define WGPU_TARGET_MACOS 1
+#define WGPU_TARGET_LINUX 2
+#define WGPU_TARGET_WINDOWS 3
+
+#define BLEND_STATES_LENGTH (1)
+#define ATTACHMENTS_LENGTH (1)
+#define RENDER_PASS_ATTACHMENTS_LENGTH (1)
+#define BIND_GROUP_LAYOUTS_LENGTH (1)
+
+WGPUByteArray read_file(const char *name) {
+    FILE *file = fopen(name, "rb");
+    fseek(file, 0, SEEK_END);
+    long length = ftell(file);
+    unsigned char *bytes = malloc(length);
+    fseek(file, 0, SEEK_SET);
+    fread(bytes, 1, length, file);
+    fclose(file);
+    return (WGPUByteArray){
+        .bytes = bytes,
+        .length = length,
+    };
+}
+
+void ReadBufferMap(
+	WGPUBufferMapAsyncStatus status, const uint8_t *data, uint8_t *userdata) {
+    if (status == WGPUBufferMapAsyncStatus_Success) {
+        printf("Times: %s \n", data);
+    }
+}
+
+int main(int argc, char *argv[]) {
+
+    if (argc != 5) {
+        return 0;
+    }
+
+    int numbers[] = {
+        atoi(argv[1]),
+        atoi(argv[2]),
+        atoi(argv[3]),
+        atoi(argv[4]),
+    };
+
+    int size = sizeof(numbers);
+
+	uint32_t numbers_length = sizeof(numbers) / sizeof(int);
+
+    WGPUInstanceId instance = wgpu_create_instance();
+
+    WGPUAdapterId adapter = wgpu_instance_get_adapter(instance,
+        &(WGPUAdapterDescriptor){
+            .power_preference = WGPUPowerPreference_LowPower,
+        });
+
+    WGPUDeviceId device = wgpu_adapter_request_device(adapter,
+        &(WGPUDeviceDescriptor){
+            .extensions =
+                {
+                    .anisotropic_filtering = false,
+                },
+        });
+
+    WGPUBufferId staging_buffer = wgpu_device_create_buffer_mapped(device,
+        &(WGPUBufferDescriptor){.size = numbers_length,
+            .usage = WGPUBufferUsage_MAP_READ | WGPUBufferUsage_TRANSFER_DST |
+                WGPUBufferUsage_TRANSFER_SRC},
+        numbers);
+
+    WGPUBufferId storage_buffer = wgpu_device_create_buffer(device,
+        &(WGPUBufferDescriptor){.size = size,
+            .usage = WGPUBufferUsage_STORAGE | WGPUBufferUsage_TRANSFER_DST |
+                WGPUBufferUsage_TRANSFER_SRC});
+
+    WGPUBindGroupLayoutId bind_group_layout =
+        wgpu_device_create_bind_group_layout(device,
+            &(WGPUBindGroupLayoutDescriptor){
+                .bindings = &(WGPUBindGroupLayoutBinding){.binding = 0,
+                    .visibility = WGPUShaderStage_COMPUTE,
+                    .ty = WGPUBindingType_StorageBuffer}});
+
+    WGPUBindGroupId bind_group = wgpu_device_create_bind_group(device,
+        &(WGPUBindGroupDescriptor){.layout = bind_group_layout,
+            .bindings = &(WGPUBindGroupBinding){
+				.binding = 0
```

Comment:
I believe `storage_buffer` is meant to be used here on the `.resource` field?

Reply:
You are right, I tried using the following code:
```rust
    WGPUBindGroupId bind_group = wgpu_device_create_bind_group(device,
        &(WGPUBindGroupDescriptor){.layout = bind_group_layout,
            .bindings = &(WGPUBindGroupBinding){.binding = 0,
                .resource =
                    &(WGPUBindingResource){.tag = WGPUBindingResource_Buffer,
                        .buffer = &storage_buffer}}});
```
This however is giving me the following compile error:
`'initializing': cannot convert from 'WGPUBindingResource *' to 'WGPUBindingResource_Tag'`

You can always send me a message in Gitter to discuss.

Reply:
`.buffer` expects to be a struct of `WGPUBindingResource_WGPUBuffer_Body`, so you need one more level (although there's an open issue about this to simplify it, see #219).

Also `.resource` expects to be a struct of `WGPUBindingResource`, not `WGPUBindingResource*`, so you can remove the extra `&` there too.

    WGPUBindGroupId bind_group = wgpu_device_create_bind_group(device,
        &(WGPUBindGroupDescriptor){.layout = bind_group_layout,
            .bindings = &(WGPUBindGroupBinding){.binding = 0,
                .resource =
                    (WGPUBindingResource){.tag = WGPUBindingResource_Buffer,
                        .buffer = (WGPUBindingResource_WGPUBuffer_Body) { storage_buffer }}}});

We could probably put some of this into an intermediate `resource` variable or similar to make it a bit clearer what's happening here :)